plugin/src/main/java/com/denizenscript/denizen/objects/EntityTag.java

```diff
@@ -3874,12 +3874,13 @@ public void adjust(Mechanism mechanism) {
         // Makes a player-type entity interact with a block.
         // -->
         if (mechanism.matches("interact_with") && mechanism.requireObject(LocationTag.class)) {
-            if (!isPlayer()) {
+            Player player = getPlayer();

```

Comment:
as was noted on Discord, the old version of this code was wrong and should probably be corrected if editing it -- should be `(if entity instanceof Player)`, ie explicitly allow player-type NPCs, which are accidentally excluded here